### PR TITLE
SPLICE-1701: JXM mbean server for cache and enginedriver exec service

### PR DESCRIPTION
**Pull request by Ben-Epstein**

### CODE REVIEW

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/GenericManagedCacheIFace.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.splicemachine.db.impl.sql.catalog;
```
Comment (jleach4) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/GenericManagedCacheIFace.java]: javadoc

Comment (jleach4) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/GenericManagedCacheIFace.java]: License Header

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/ManagedCacheMBean.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package com.splicemachine.db.impl.sql.catalog;
```
Comment (jleach4) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/ManagedCacheMBean.java]: License header

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/ManagedCacheMBean.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import javax.management.MXBean;
+import java.io.Serializable;
+import java.util.WeakHashMap;
+
+@MXBean
```
Comment (jleach4) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/ManagedCacheMBean.java]: javadoc

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/TotalManagedCache.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
+package com.splicemachine.db.impl.sql.catalog;
```
Comment (jleach4) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/TotalManagedCache.java]: licensed header

File: splice_machine/src/main/java/com/splicemachine/EngineDriver.java [hunk 5]
```diff
@@ -96,7 +130,7 @@ public void close(SpliceSequence entity) throws Exception{
 
         /* Create a general purpose thread pool */
         final AtomicLong count = new AtomicLong(0);
-        this.threadPool = new ThreadPoolExecutor(20, config.getThreadPoolMaxSize(),
+        ThreadPoolExecutor tpe = new ThreadPoolExecutor(0, config.getThreadPoolMaxSize(),
```
Comment (jleach4) [inline splice_machine/src/main/java/com/splicemachine/EngineDriver.java]: Why did this change?

File: splice_machine/src/main/java/com/splicemachine/EngineDriver.java [hunk 6]
```diff
@@ -164,4 +199,176 @@ public OlapClient getOlapClient() {
     public ExecutorService getExecutorService() {
         return threadPool;
     }
+
+
+    private static final ResultColumnDescriptor[] EXEC_SERVICE_COLUMNS= {
+            new GenericColumnDescriptor("host",DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR)),
```
Comment (jleach4) [inline splice_machine/src/main/java/com/splicemachine/EngineDriver.java]: Capitalize H

File: splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceAdminIT.java [hunk 7]
```diff
@@ -238,6 +224,41 @@ public void testGetWriteIntakeInfo() throws Exception {
     }
 
     @Test
+    public void testGetExecInfo() throws Exception {
+        CallableStatement cs = methodWatcher.prepareCall("call SYSCS_UTIL.SYSCS_GET_EXEC_SERVICE_INFO()");
+        ResultSet rs = cs.executeQuery();
+        TestUtils.FormattedResult fr = TestUtils.FormattedResult.ResultFactory.convert("call SYSCS_UTIL.SYSCS_GET_EXEC_SERVICE_INFO()", rs);
+        System.out.println(fr.toString());
```
Comment (jleach4) [inline splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceAdminIT.java]: Remove System.out

Comment (jleach4) [inline splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceAdminIT.java]: What are you testing?

File: splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceAdminIT.java [hunk 8]
```diff
@@ -367,4 +388,20 @@ public void testSplitTableTableDoesNotExist() throws Exception {
...
+        SpliceUnitTest.MyWatcher tableWatcher =
+                new SpliceUnitTest.MyWatcher(TABLE_NAME,CLASS_NAME,
+                        "(PARCELID INTEGER UNIQUE NOT NULL, ADDRESS VARCHAR(15), BOARDDEC VARCHAR(11), EXSZONE VARCHAR(8), PRPZONE VARCHAR(8), HEARDATE DATE)");
+        tableDAO.drop(CLASS_NAME, TABLE_NAME);
+        tableWatcher.create(Description.createSuiteDescription(CLASS_NAME, "testGetSchemaInfo"));
+
+        CallableStatement cs = methodWatcher.prepareCall("call SYSCS_UTIL.SYSCS_GET_SCHEMA_INFO()");
+        ResultSet rs = cs.executeQuery();
+        TestUtils.FormattedResult fr = TestUtils.FormattedResult.ResultFactory.convert("call SYSCS_UTIL.SYSCS_GET_SCHEMA_INFO()", rs);
+        System.out.println(fr.toString());
```
Comment (jleach4) [inline splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceAdminIT.java]: Remove System.out

File: splice_machine/src/test/test-data/test_data/bad_import/db1082.csv [hunk 9]
```diff
@@ -1 +0,0 @@
-1,1
```
Comment (jleach4) [inline splice_machine/src/test/test-data/test_data/bad_import/db1082.csv]: ?

File: splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceAdminIT.java (hunk 7 above)
Comment (Ben-Epstein) [inline splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceAdminIT.java]: The print for the FormattedResult is in all tests in this class

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/GenericManagedCacheIFace.java (hunk 1 above)
Comment (Ben-Epstein) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/GenericManagedCacheIFace.java]: resolved

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/ManagedCacheMBean.java (hunk 2 above)
Comment (Ben-Epstein) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/ManagedCacheMBean.java]: resolved

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/GenericManagedCacheIFace.java (hunk 1 above)
Comment (Ben-Epstein) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/GenericManagedCacheIFace.java]: resolved

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/TotalManagedCache.java (hunk 4 above)
Comment (Ben-Epstein) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/TotalManagedCache.java]: resolved

Comment (Ben-Epstein) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/TotalManagedCache.java]: resolved

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/ManagedCacheMBean.java (hunk 2 above)
Comment (Ben-Epstein) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/ManagedCacheMBean.java]: resolved

**Comment by Ben-Epstein** [comment]
@jleach4 changes resolved

File: splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java [hunk 10]
```diff
@@ -164,6 +164,19 @@ public static void SYSCS_GET_WRITE_INTAKE_INFO(ResultSet[] resultSets) throws SQ
         PipelineAdmin.SYSCS_GET_WRITE_INTAKE_INFO(resultSets);
     }
 
+    public static void SYSCS_GET_EXEC_SERVICE_INFO(ResultSet[] resultSets) throws SQLException{
```
Comment (Ben-Epstein) [inline splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java]: Made changes, am offline, will push when I am connected

**Pull request closed without merging**